### PR TITLE
Introducing Versioning to docs

### PR DESCRIPTION
**Pull request by gusfcarvalho**
This PR implements versioning to docs using **mike**.

~Draft~ PR Ready for Review. It does the following:

- [x] generate `main` docs for each merge to main
- [x] generate release tags for each release and tag the latest one as `latest` 
- [x] generate a nice "You are viewing old docs" message whenever we are on a `non-latest` doc

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
I am having troubles to use the `overrides` folder from [here](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#version-warning). It just seems to keep ignoring anything I put there.

Any of you have already worked with it?

**Comment by gusfcarvalho** [comment]
Also, do you think it is better to clean up any local changes made with `mike deploy` within the makefile itself?

File: hack/api-docs/Makefile [hunk 1]
```diff
@@ -45,13 +48,23 @@ image:
...
+.PHONY: build.publish
+build.publish: image generate $(SOURCES)
+	mkdir -p $(GENROOT)
+	$(DOCKER) run \
+	    --mount type=bind,source=$(ROOT),target=/repo \
+		--sig-proxy=true \
+		--rm \
+		--user $(UID):$(GID) \
+		$(MKDOCS_IMAGE) \
+		/bin/bash -c "cd /repo && git config user.name "Docs" && git config user.email "docs@external-secrets.io" && $(MIKE) deploy --update-aliases -p -F hack/api-docs/mkdocs.yml $(DOCS_VERSION) $(DOCS_ALIAS);"
```
Comment (moolen) [inline hack/api-docs/Makefile:67]: We can move the `git config --global user.(name|email)` to the Dockerfile to keep it a bit simpler

File: hack/api-docs/mkdocs.yml [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 repo_url: https://github.com/external-secrets/external-secrets
 repo_name: External Secrets Operator
-site_dir: site
-docs_dir: docs
+site_dir: ../../site
+docs_dir: ../../docs
+remote_branch: gh-pages
 theme:
   name: material
+  custom_dir: ../../overrides
```
Comment (moolen) [inline hack/api-docs/mkdocs.yml:9]: i cross-checked it with the docs and other projects and it seems correctly configured. Could be that we have to use paths relative to the (maybe?) docs dir. Does it not change at all even if replacing `main.html` with junk?

File: hack/api-docs/Makefile (hunk 1 above)
Comment (gusfcarvalho) [reply hack/api-docs/Makefile:67]: Will do! I think I understood what was missing for the `not seeing the latest` banner. I'm going to test it, and in the following I'll move the git config.

File: hack/api-docs/mkdocs.yml (hunk 2 above)
Comment (gusfcarvalho) [reply hack/api-docs/mkdocs.yml:9]: It seems it was some version bump that was needed. I'm testing it right now

**Comment by gusfcarvalho** [comment]
"Seeing not the latest docs":
image

"Seeing Latest docs":
image

I didn't want to test the release thingy in github actions without a proper merge and review (as it would create a new release)

File: hack/api-docs/mkdocs.yml (hunk 2 above)
Comment (gusfcarvalho) [reply hack/api-docs/mkdocs.yml:9]: Indeed that was it!

File: hack/api-docs/Makefile (hunk 1 above)
Comment (gusfcarvalho) [reply hack/api-docs/Makefile:67]: it seems my latest commit is not being taken into account by github... but changes are [here](https://github.com/external-secrets/external-secrets/pull/663/commits/ade03c6a03f44c4a2db09ad8326c0ded20736a84)

Comment (moolen) [reply hack/api-docs/Makefile:67]: The publish target still has the `git config` things

**Comment by gusfcarvalho** [comment]
I just realized I created this branch from `fix/docs`, so I also merged it with main to avoid any weird conflict it might have had.

File: hack/api-docs/Makefile (hunk 1 above)
Comment (gusfcarvalho) [reply hack/api-docs/Makefile:67]: forgot about this one :s

**Pull request merged**